sidechain/top-pool-executor/src/getter_operator.rs

```diff
@@ -0,0 +1,122 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	error::{Error, Result},
+	TopPoolOperationHandler,
+};
+use codec::Encode;
+use ita_stf::{hash::TrustedOperationOrHash, TrustedGetterSigned};
+use itp_stf_executor::traits::{StfExecuteTimedCallsBatch, StfExecuteTimedGettersBatch};
+use itp_types::{ShardIdentifier, H256};
+use its_primitives::traits::{Block as SidechainBlockT, SignedBlock as SignedBlockT};
+use its_state::{SidechainState, SidechainSystemExt, StateHash};
+use its_top_pool_rpc_author::traits::{AuthorApi, OnBlockCreated, SendState};
+use log::*;
+use sgx_externalities::SgxExternalitiesTrait;
+use sp_runtime::{traits::Block as BlockT, MultiSignature};
+use std::{time::Duration, vec, vec::Vec};
+
+/// Interface to the trusted getters within the top pool
+pub trait TopPoolGetterOperator {
+	/// Loads trusted getters from the top pool for a given shard
+	/// and executes them until either all calls are executed or `max_exec_duration` is reached.
+	fn execute_trusted_getters_on_shard(
+		&self,
+		shard: &ShardIdentifier,
+		max_exec_duration: Duration,
+	) -> Result<()>;
+
+	/// Retrieves trusted getters from the top pool.
+	fn get_trusted_getters(&self, shard: &ShardIdentifier) -> Result<Vec<TrustedGetterSigned>>;
+
+	/// Removes the given trusted getter from the top pool.
+	/// FIXME: Hash type should be taken from TrustedGetter itself: #515
```

Comment:
Does it return the hash of the removed getter?

Reply:
Yes.